api/zeebe/zeebe-openapi.yaml

```diff
@@ -0,0 +1,386 @@
+          type: string
+          nullable: false
+        allowOverride:
+          description: >
+            By default, the task is reassigned if it was already assigned. Set this to `false`
+            to return an error in such cases. The task must then first be unassigned to
+            be assigned again. Use this when you have users picking from group task
+            queues to prevent race conditions.
+          type: boolean
+          nullable: true
+        action:
+          description: >
+            A custom action value that will be accessible from user task events resulting
+            from this endpoint invocation. If not provided, it will default to "assign".
+          type: string
+          nullable: true
+    UserTaskUpdateRequest:
+      type: object
+      properties:
+        changeset:
+          description: |
+            JSON object with changed task attribute values.
+
+            The following attributes can be adjusted with this endpoint, additional attributes
+            will be ignored:
+
+            * `candidateGroups` - reset by providing an empty list
+            * `candidateUsers` - reset by providing an empty list
+            * `dueDate` - reset by providing an empty String
+            * `followUpDate` - reset by providing an empty String
+
+            Providing any of those attributes with a `null` value or omitting it preserves
+            the persisted attribute's value.
+
+            The assignee cannot be adjusted with this endpoint, use the Assign task endpoint.
+            This ensures correct event emission for assignee changes.
+          type: object
+          nullable: true
+          $ref: "#/components/schemas/Changeset"
+        action:
+          description: >
+            A custom action value that will be accessible from user task events resulting
+            from this endpoint invocation. If not provided, it will default to "update".
+          type: string
+          nullable: true
+    Variables:
+      description: A map of variables.
+      type: object
+      additionalProperties: true
+    Changeset:
+      description: A map of changes.
+      type: object
+      additionalProperties: true
+      properties:
+        dueDate:
+          type: string
+          description: The due date of the task. Reset by providing an empty String.
+          nullable: true
+        followUpDate:
+          type: string
+          description: The follow-up date of the task. Reset by providing an empty String.
+          nullable: true
+        candidateUsers:
+          type: array
+          description: The list of candidate users of the task. Reset by providing an empty list.
+          items:
+            type: string
+          nullable: true
+        candidateGroups:
+          type: array
+          description: The list of candidate groups of the task. Reset by providing an empty list.
+          items:
+            type: string
+          nullable: true
```

Comment:
❓ @pepopowitz, any ideas why the additional properties here have the description from the parent object `UserTaskUpdateRequest` again?
![image](https://github.com/camunda/camunda-docs/assets/586643/4a62f8c0-9eac-4309-bce0-ac1aeff0b842)

Reply:
💭 The same happens for the `variables` in the user task completion endpoint
![image](https://github.com/camunda/camunda-docs/assets/586643/5a02679a-6ff3-4152-8d71-68344c606be7)

Reply:
It looks like the `additionalProperties: true` property on the schema object not only renders the `property/name*` property, but uses the same description as the schema object for the additional properties item. 

I'll see if I can figure out a way to avoid that, whether it's something I can override in the plugin or something I can change in the schema.

Reply:
I pushed up [a fix for this](https://github.com/camunda/camunda-docs/pull/3487/commits/1a27d42025a5ee20b36c173e947d2785cffd6771). I haven't yet deployed it to the preview site. 

## The problem

The plugin we're using to generate these docs handles `additionalProperties: true` in a funny way. It _does_ emit the `property *name` map, but it also duplicates the description from the parent schema.

I looked into multiple ways to work around this with alternative `additionalProperties` formats -- each of which displays in a way that makes sense using https://editor.swagger.io/, so I know it would be a valid schema -- but the plugin didn't emit any of these in a way that made sense. It seems to have limited support for the `additionalProperties` property.

## The workaround

Instead, I chose to swizzle (override) a rendering component from the OpenAPI plugin, to scrub the duplicate description in a context where it looks like the `additionalProperties` map is being rendered. Luckily for us, the `name` of the item being rendered is rather unique in these conditions. 

Here's what the two schemas in question now look like: 

![image](https://github.com/camunda/camunda-docs/assets/1627089/714e592b-85b5-4317-9c36-639e93c7731d)

![image](https://github.com/camunda/camunda-docs/assets/1627089/5b47cf26-94da-4687-91ed-02439aca5c1e)

And here's what https://editor.swagger.io/ displays for these properties, to demonstrate alignment: 

![image](https://github.com/camunda/camunda-docs/assets/1627089/e0d7e6c0-5067-4357-b3fa-f28af21cb3b7)

![image](https://github.com/camunda/camunda-docs/assets/1627089/7ead9d5f-4087-47c3-b088-b60d3ef57d7b)


## Does the message `Additional properties allowed." capture everything?

If there's a message that would make more sense in both situations, let me know! I chose this message because it aligns with how Swagger displays it. I'd rather not override the messages on a per-field basis.

Reply:
Great fix, @pepopowitz, I like the alignment with what Swagger displays 👍